lib/sidekiq/middleware/server/active_record.rb

```diff
@@ -2,6 +2,21 @@ module Sidekiq
   module Middleware
     module Server
       class ActiveRecord
+        def initialize
+          # We can't use this middleware with the Rails reloader.
+          #
+          # The reloader needs the active connection to clear its query cache
+          # before releasing it otherwise other workers will see dirty query
+          # cache data.
+          #
+          # We need to make sure folks remove this middleware if they've added
+          # it themselves.
```

Comment:
Probably worth an explicit acknowledgement that this middleware is unnecessary because the reloader will do the job for us -- that it's not "incompatible so you're trading one feature for the other", it's "incompatible because duplication"